server/src/index.ts

```diff
@@ -1,16 +1,87 @@
 import express, { Request, Response } from "express";
-import { GameState } from "shared";
+import { GameState, SERVER_PORT, PlayerId, Card, NUMBER_DIGIT_CARDS_IN_GAME, ALL_DIGIT_CARDS, NUMBER_OPERATION_CARDS_IN_GAME, ALL_OPERATION_CARDS, isExtensionOfExpression} from "shared";
+import http from "http";
+import {Server} from "socket.io";
+
 
 const app = express();
-// It would be nicer if it was defined in shared/constants 
-const port = 3001;
+const server = http.createServer(app);
+const io = new Server(server, {
+    cors : {
+        origin : "http://localhost:5173" 
+    }
+});
+
+
+const port = SERVER_PORT;
+
+
+const current_games = new Map(); //A dict with current games {id : game}
+let last_game_id = 0; //Last game id (to get a new one)
+
+let players_queue : PlayerId[] = []; //A queue of current waiting players (id)
+
 
 // Define the root path with a greeting message
 app.get("/", (req: Request, res: Response) => {
     res.json({ message: "Welcome to the Express + TypeScript Server!" });
 });
 
 // Start the Express server
-app.listen(port, () => {
+server.listen(port, () => {
     console.log(`The server is running at http://localhost:${port}`);
-});
+}); 
+
+
+
+//Create a game for two players and return an instance of game state
+function createGame(player_id_1 : PlayerId, player_id_2 : PlayerId) : GameState{
+
+    let cards : Card[] = [];
+    //get a random cards 
+    for (let nr_card = 0; nr_card < NUMBER_DIGIT_CARDS_IN_GAME; nr_card++){
+        cards.push(ALL_DIGIT_CARDS[Math.floor(Math.random() * ALL_DIGIT_CARDS.length)]);
+    }
+
+    for (let nr_card = 0; nr_card < NUMBER_OPERATION_CARDS_IN_GAME; nr_card++){
+        cards.push(ALL_OPERATION_CARDS[Math.floor(Math.random() * ALL_OPERATION_CARDS.length)]);
+    }
+
+
+    //Init a game
+    let game : GameState = {player1Id : player_id_1, player2Id : player_id_2, player1exp : [1], player2exp : [1], currentTurn : (Math.floor(Math.random() * 2) + 1 as 1 | 2), availableCards : cards};
```

Comment:
Ja bym to w ogóle przeniósł do całe do funkcji `initGameState` która może żyć w pliku `game_state.ts` w shared.